### ISSUE TITLE
Can you share your code of preparing the data?

### ISSUE THREAD

**Issue by wuqiangch** [ZhiwenShao/dense-landmark-location#3]
@ZhiwenShao How to prepare the trainding data? Can you share your code of preparing the data? Thanks!!!

**Comment by ZhiwenShao** [comment]
Our data augmentation method consist of four operations: rotation, translation, horizontal flip and JPEG compression. For each training image, you can first conduct rotation, and determine face bounding boxes based on new generated landmarks. Further you may perform translation, horizontal flip and JPEG compression. We suggest you using the relevant functions of OpenCV to implement. Good Luck!

**Comment by wuqiangch** [comment]
@ZhiwenShao Thank! Did you use  all the training data to generate the mean data ? Or other? The training data only need to minus the mean data? The groudtrueth of landmarks should to devide the face's width and heigt? Thank!!

**Comment by ZhiwenShao** [comment]
We use all the training data to generate the mean data. The landmarks don't need to be divided, with using normalized Euclidean loss.

**Comment by wuqiangch** [comment]
@ZhiwenShao Thanks！The training data only need to minus the mean data？  In the  train_val.prptotxt,the "ground_split1" means Principal subset , the "ground_split2"means Elaborate subset and the "biocular" means all the landmarks,Is it right? Thank!!

**Comment by ZhiwenShao** [comment]
Yes, you can also use the scale parameter besides minus the mean.
Yes.

**Comment by wuqiangch** [comment]
@ZhiwenShao Good，Thanks very much！！

**Comment by ZhiwenShao** [comment]
You're welcome!

**Comment by wuqiangch** [comment]
@ZhiwenShao 
![image](https://cloud.githubusercontent.com/assets/7983007/23747182/bdfec064-04f9-11e7-8a6e-e900f0fd588d.png)
what's wrong with this? The loss is less than zero?

**Comment by ZhiwenShao** [comment]
There must be some errors with your input training data. Maybe you need to check it.

**Comment by wuqiangch** [comment]
@ZhiwenShao  I found the problem. But I alse get a bad model. Can you share your trained model?

**Comment by ZhiwenShao** [comment]
You can test the models MCNet (https://github.com/ZhiwenShao/MCNet) which is a improvement version of CFT.
